packages/block-library/src/image/edit.native.js

```diff
@@ -631,9 +631,13 @@ export class ImageEdit extends Component {
 		const additionalImageProps = {
 			height: '100%',
 			resizeMode: context?.imageCrop ? 'cover' : 'contain',
+			hideImageCaption: context?.hideImageCaption,
+			fixedHeight: context?.fixedHeight,
 		};
 
-		const imageContainerStyles = [ hasImageContext && styles.fixedHeight ];
+		const imageContainerStyles = [
```

Comment:
👋  @geriux, I wanted to check in with you, as this PR makes some changes to the ones you introduced with your recent PR in https://github.com/WordPress/gutenberg/pull/34957. 

The reason for the changes is that `hasImageContext` was evaluating to `true` for the Tiled Gallery block, and the block needs to have more flexible heights. You'll see from this PR that I've worked around this by adding a new `fixedHeight` context object to the Gallery block, so it's only used with that block. 

Does that seem like an okay approach to you? Interested to know whether you have other thoughts.

Reply:
Since the only block that was using context was the Gallery block I used a generic `fixedHeight` name to avoid limiting the height of the component.

Seeing that now the Tiled block would need this as well, what if we use a name like `isGallery` instead of `fixedHeight` for the context?

Do we also need to pass `fixedHeight` from the Image block?

Reply:
@geriux, thank you for looking into this! 🙇‍♀️ 

> Seeing that now the Tiled block would need this as well, what if we use a name like `isGallery` instead of `fixedHeight` for the context?

Ah, the Tiled Gallery block doesn't need this context. I created it as `hasImageContext` was evaluating as true for the Tiled Gallery block and applying the fixed height that was needed only for the core Gallery block. This was preventing images from being styled as squares within the Tiled Gallery block. That's what led me to create a context that would only apply to the core Gallery block, not other blocks that are using contexts.

I'm happy to rename it if you want it be clearer that this is for use with the core gallery, maybe something like `galleryHasFixedHeight` or `isCoreGallery`?

> Do we also need to pass `fixedHeight` from the Image block?

That's a good point! It doesn't really need to be passed as an `additionalImageProp`. Do you think a new `imageBlockStyles` object would make sense here?

Reply:
I've gone ahead to create an `imageBlockStyles` object and moved the `fixedHeight` context to it in https://github.com/WordPress/gutenberg/pull/35089/commits/ce9ec06e1c0e5b3a75a2a985f2f127a472d482ee.